### PR TITLE
Throw an error when mlir's outpus shape is non-standard

### PR DESCRIPTION
**Pull request by pfultz2**
This is currently supported in mlir. This can be removed in the future when mlir is updated.

### CODE REVIEW

File: src/targets/gpu/fuse_mlir.cpp [hunk 1]
```diff
@@ -103,7 +103,9 @@ struct mlir_op
             }
             if(ins->name() == "@return")
             {
-                return ins_shapes[ins->inputs().at(0)].with_type(type);
+                auto s = ins_shapes[ins->inputs().at(0)].with_type(type);
```
Comment (umangyadav) [inline src/targets/gpu/fuse_mlir.cpp:106]: can the output be a tuple type ?

Comment (pfultz2) [reply src/targets/gpu/fuse_mlir.cpp:106]: Not currently.

**Review by umangyadav — APPROVED** [review]

**Comment by pfultz2** [comment]
Fixes: #2144.

**Comment by pfultz2** [comment]
> Will migraphx be able to catch this exception and recover when running nhwc?

Yes because we first insert contiguous operators and then eliminate_contiguous will remove contiguous if there is no exception.

**Comment by krzysz00** [comment]
Wait, does eliminate_contiguous run before or after fuse_mlir? If after ... how does it hit this exception?

**Comment by pfultz2** [comment]
It runs after lowering.

**Pull request merged**